nx/lib/nx/binary_backend/matrix.ex

```diff
@@ -237,44 +237,29 @@ defmodule Nx.BinaryBackend.Matrix do
     {l, u} =
       for j <- 0..(n - 1), reduce: {zeros_matrix, zeros_matrix} do
         {l, u} ->
-          i_range = if j == 0, do: [0], else: 0..j
-          [l_row_j] = get_matrix_rows(l, [j])
-          l = replace_rows(l, [j], [replace_vector_element(l_row_j, j, 1.0)])
-
-          u_col_j =
-            for i <- i_range,
-                reduce: get_matrix_column(u, j) do
-              u_col_j ->
-                l_row_slice = slice_matrix(l, [i, 0], [1, i])
-
-                u_col_slice = slice_vector(u_col_j, 0, i)
-
-                s = dot_matrix(l_row_slice, u_col_slice)
-                [a_elem] = get_matrix_elements(a_prime, [[i, j]])
-                replace_vector_element(u_col_j, i, a_elem - s)
+          l = replace_matrix_element(l, j, j, 1.0)
+
+          u =
+            for i <- 0..j, reduce: u do
+              u ->
+                u_slice = slice_matrix(u, [0, j], [i, 1])
+                l_slice = slice_matrix(l, [i, 0], [1, i])
+                sum = dot_matrix(u_slice, l_slice)
+                [a_ij] = get_matrix_elements(a_prime, [[i, j]])
+                replace_matrix_element(u, i, j, a_ij - sum)
             end
 
-          u = replace_cols(u, [j], transpose_matrix(u_col_j))
-
-          [u_jj] = get_matrix_elements(u, [[j, j]])
-
           l =
-            if u_jj < eps do
-              l
-            else
-              i_range = if j == n - 1, do: [n - 1], else: j..(n - 1)
-
-              for i <- i_range, reduce: l do
-                l ->
-                  u_col_slice = slice_matrix(u, [0, j], [j, 1])
-                  [l_row_i] = get_matrix_rows(l, [i])
-                  s = dot_matrix(u_col_slice, l_row_i)
-                  [a_elem] = get_matrix_elements(a_prime, [[i, j]])
+            for i <- j..(n - 1), i != j, reduce: l do
```

Comment:
update/facepalm: It seems that the error was that `if u_jj < eps` should be `if abs(u_jj) < eps`
However, the new implementation is cleaner (albeit a little bit less efficient) and I want to keep it for readability.